practices/follow-functional-core-imperative-shell.md

```diff
@@ -3,117 +3,153 @@
+Integration tests should cover the behavior of the imperative shell as it interacts with external systems, including database calls, API requests, or user interfaces.
+Imperative shell integration tests ideally require fewer scenarios to validate, given that control statements such as `if`, `while`, or `for` loops should mostly reside within the functional core layer.
+
+### Epecially beneficial when using dynamic languages
+
+<ed> This sounds like a benefit, not a challenge/pitfall/limitation. Consider moving this note above to the paragraph introducing the Practice. </ed>
 
-### Is Specially beneficial when using dynamic languages
+In dynamic languages, the Functional Core, Imperative Shell approach is often favored because it can address testing challenges inherent in these languages.
+By communicating between the functional core and imperative shell via passing values instead of relying on interfaces, developers can mitigate the risk of false positives when using mocks and avoid encountering errors with the real implementation.
 
-In dynamic languages, the Functional Core, Imperative Shell approach is often favored due to its ability to address testing challenges inherent in these languages.
-By communicating between the Functional Core and Imperative Shell via passing values instead of relying on interfaces, developers can mitigate the risk of false positives when using mocks and encountering errors with the real implementation.
 
 ## How to Improve
 
-### [Host A Viewing Party](/practices/host-a-viewing-party.md)
+<ed> Ditto comment in other Practices about renaming this section to be a bit more specific (e.g., "How to Implement This Practice" or "Tips for Implementing This Practice"). </ed> 
 
-#### [Boundaries](https://www.destroyallsoftware.com/talks/boundaries)
+Help your team implement this practice by hosting a viewing party of specially curated talks, starting a book club to discuss impactful titles on this topic, or facilitating a roundtable discussion about relevant topics.
 
-This talk is about using simple values (as opposed to complex objects) not just for holding data, but also as the boundaries between components and subsystems. It moves through many topics: functional programming; mutability's relationship to OO; isolated unit testing with and without test doubles; and concurrency, to name some.
 
-#### [Moving IO to the edges of your app: Functional Core, Imperative Shell](https://www.youtube.com/watch?v=P1vES9AgfC4)
+### [Host a Viewing Party](/practices/host-a-viewing-party.md)
+
+#### [Boundaries by Gary Bernhardt](https://www.destroyallsoftware.com/talks/boundaries)
+
+This talk is about using simple values (as opposed to complex objects) not just for holding data but also as the boundaries between components and subsystems. It moves through many topics: functional programming, mutability's relationship to OOP, isolated unit testing with and without test doubles, concurrency, and more.
```

Comment:
let's kill the `not just for holding data but also` part.

Reply:
Will do.